pkg/ha/filter_test.go

```diff
@@ -356,13 +380,6 @@ func TestHaParserParseData(t *testing.T) {
 				t.Fatalf("wanted error from FilterData, got nil")
 			}
 
```

Comment:
we aren't testing c.proceed anymore are we missing any important cases?

Reply:
proceed was removed because the signature changed. now we just empty the write request if we don't want to proceed. simplifies the interfaces and we were already checking for empty write requests.